### PR TITLE
feat: INSTR function #881

### PR DESCRIPTION
**Pull request by mateuszmrozewski**
Implements #881

### Description 
This adds implementation of INSTR function as described in https://docs.oracle.com/database/121/SQLRF/functions089.htm#SQLRF00651 (reported in #881)

### Testing done 
I have added unit tests and did manual testing.

Please note that I am looking for initial feedback only. I am aware that functional tests and documentation needs to be added.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by mateuszmrozewski** [comment]
[clabot:check]

**Comment by mateuszmrozewski** [comment]
@agavra, I have added functional tests and documentation. I have also added a few unit test cases that shown an issue with the initial implementation. Please advise if there is anything else that needs to be added.

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 1]
```diff
@@ -316,6 +317,30 @@ INITCAP
 Capitalize the first letter in each word and convert all other letters
 to lowercase. Words are delimited by whitespace.
 
+INSTR
+-----
+
+`INSTR(string, substring, [position], [occurrence])`
+
+Returns a position of `substring` in `string` (first character is at position 1).
+If `position` is given search starts from that position.
```
Comment (agavra) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:326]: we should specify here what it means to have negative position

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/Instr.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import static org.apache.commons.lang3.StringUtils.substring;
+
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.util.KsqlConstants;
+import org.apache.commons.lang3.StringUtils;
+
+@UdfDescription(name = "instr",
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "PROPER DESCRIPTION")
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/Instr.java:28]: let's fill this out 😂

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 3]
```diff
@@ -316,6 +317,30 @@ INITCAP
 Capitalize the first letter in each word and convert all other letters
 to lowercase. Words are delimited by whitespace.
 
+INSTR
+-----
+
+`INSTR(string, substring, [position], [occurrence])`
+
+Returns a position of `substring` in `string` (first character is at position 1).
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:325]:
```suggestion
Returns the position of `substring` in `string`. The first character is at position 1.
```

File: docs/developer-guide/ksqldb-reference/scalar-functions.md (hunk 1 above)
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:326]:
```suggestion
If `position` is provided, search starts from the specified position. 
```

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 4]
```diff
@@ -316,6 +317,30 @@ INITCAP
...
+If `position` is given search starts from that position. 
+If `occurrence` is given position of n-th occurrence is returned.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:327]:
```suggestion
If `occurrence` is provided, the position of *n*-th occurrence is returned.
```

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 5]
```diff
@@ -316,6 +317,30 @@ INITCAP
...
+
+Negative `position` causes the search to work from end to start of `string`.
+
+If `substring` is not found the return value is 0.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:331]:
```suggestion
If `substring` is not found, the return value is 0.
```

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/Instr.java (hunk 2 above)
Comment (mateuszmrozewski) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/Instr.java]: :)

**Comment by mateuszmrozewski** [comment]
All suggestions applied. I have also committed the plans, specs and topologies for functional tests.

**Pull request merged**